File: 2018 July 15/snek.cpp
#include <iostream>
using namespace std;

main()
{
  int n; cin >> n;
  int sum;

  for(int i = 0; i < n; i++)
  {
    for(int j = 0; j < i+1; j++)
    {
      sum = 0;
      for(int k = n; k > 5 - j - 1; k--) { sum += k;}

      if(j%2==0) { cout << (sum - n + i + 1) << " ";}
      else { cout << (sum - i + j) << " ";}
    }
    cout << '\n';
  }
}

File: 2018 July 15/snek.py
n = int(input())

for i in range(n):
    out = ""
    for j in range(i+1):
        j += 1

        sum = 0
        for k in range(n, 5 - j, -1):
            sum += k

        if(j%2 == 1):
            out += str(sum - n + i + 1) + " "
        else:
            out += str(sum - i + n) + " "

    print(out)
